### ISSUE TITLE
Create Linux Release

### ISSUE THREAD

**Issue by dail8859** [dail8859/NotepadNext#72]
Seems like an [AppImage](https://appimage.org/) is the best way to make easily sharable releases for Linux.

Two main packages to do this: linuxdeployqt and linuxdeploy (with linuxdeploy-plugin-qt). Since this project will eventually move to Qt6 linuxdeploy seems to be the only choice. See commit message of https://github.com/mapeditor/tiled/commit/f6a203b1e384a5d68420d2c6336135e4044800f0

Reference:
https://github.com/linuxdeploy/linuxdeploy/
https://github.com/linuxdeploy/linuxdeploy-plugin-qt/

After alot of trial and error I think I did something like this...not sure if all this is exactly needed and this can be improved. But it is at least functional enough to run.

Copy `icon/nn.svg` to the root directory as `NotepadNext.svg`. Also a desktop file needs created. I very quickly put this together:

```
[Desktop Entry]
Exec=NotepadNext
Type=Application
Icon=NotepadNext
Name="Notepad Next"
Categories=Qt;TextEditor;Development
```

Then ran...

```
wget --no-verbose "https://github.com/linuxdeploy/linuxdeploy/releases/download/continuous/linuxdeploy-x86_64.AppImage"
wget --no-verbose "https://github.com/linuxdeploy/linuxdeploy-plugin-qt/releases/download/continuous/linuxdeploy-plugin-qt-x86_64.AppImage"
chmod +x linuxdeploy*.AppImage

# Uh...I think I did this...
export OUTPUT=NotepadNext-x86_64.AppImage

./linuxdeploy-x86_64.AppImage --appdir AppDir -e build/NotepadNext/NotepadNext -d NotepadNext.desktop -i NotepadNext.svg --plugin qt --output appimage
```

Note: I see mention of [INSTALL_ROOT](https://docs.appimage.org/packaging-guide/from-source/native-binaries.html#qmake) when building applications for an AppImage but I haven't been able to get it to work and it doesn't seem to be causing issues. Maybe the .pro file need an INSTALLS target?

**Comment by FollowerOfBigboss** [comment]
Hey @dail8859, like you i have been looking packaging NotepadNext with AppImage. I managed to get AppDir by adding these lines end of `src/NotepadNext/NotepadNext.pro` file.

```
unix {
 target.path = /usr/bin
} 
INSTALLS += target
```
You will see AppDir at `build/NotepadNext/` after this instruction
`make install INSTALL_ROOT=AppDir`

**Comment by dail8859** [comment]
@FollowerOfBigboss

I've been trying to find some more mature Qt projects on Github to try to see how they do some of this stuff. I'm no expert with Qt (especially qmake) so I'm sure there are better ways to set up the build structure for Notepad Next. (e.g. I don't fully understand `DESTDIR`, `PREFIX` and some of the other stuff...but it works for now)

> I managed to get AppDir by adding these lines end of src/NotepadNext/NotepadNext.pro file

Awesome! That lines up with what I was finding as well. Looking at https://github.com/mapeditor/tiled/blob/316ceb4dfd5c5bb0ff26c4df885dd1c02d0c92b1/src/tiled/tiled.pro#L562-L604 it seems the `INSTALLS` variable would be a good use for installing things that need to go with the binary file such as the License, .desktop file, icons, etc which would allow for easier preparation for `linuxdeploy`.

**Comment by dail8859** [comment]
Closed by #75
